tests/contrib/unittest_plugin/test_unittest.py

```diff
@@ -169,29 +300,59 @@ def test_will_be_skipped_second(self):
-            assert spans[0].get_tag(COMPONENT) == COMPONENT_VALUE
-            assert spans[0].get_tag(test.TYPE) == SpanTypes.TEST
-            assert spans[0].get_tag(test.NAME) == "test_will_be_skipped_first"
-            assert spans[0].get_tag(test.MODULE) == "tests.contrib.unittest_plugin.test_unittest"
-            assert spans[0].get_tag(test.SUITE) == "UnittestExampleTestCase"
-            assert spans[0].get_tag(test.TEST_STATUS) == test.Status.SKIP.value
-            assert spans[0].get_tag(test.SKIP_REASON) == "demonstrating skipping with a reason"
-
-            assert spans[1].get_tag(test.TEST_TYPE) == SpanTypes.TEST
-            assert spans[1].get_tag(test.TEST_FRAMEWORK) == FRAMEWORK
-            assert spans[1].get_tag(SPAN_KIND) == KIND
-            assert spans[1].get_tag(COMPONENT) == COMPONENT_VALUE
-            assert spans[1].get_tag(test.TYPE) == SpanTypes.TEST
-            assert spans[1].get_tag(test.NAME) == "test_will_be_skipped_second"
-            assert spans[1].get_tag(test.MODULE) == "tests.contrib.unittest_plugin.test_unittest"
-            assert spans[1].get_tag(test.SUITE) == "UnittestExampleTestCase"
-            assert spans[1].get_tag(test.TEST_STATUS) == test.Status.SKIP.value
-            assert spans[1].get_tag(test.SKIP_REASON) == "demonstrating skipping with a different reason"
+            assert len(spans) == 5
+
+            test_suite_span = spans[2]
+            test_module_span = spans[1]
+            test_session_span = spans[0]
+
+            for i in range(len(spans)):
+                assert spans[i].get_tag(test.TEST_TYPE) == SpanTypes.TEST
+                assert spans[i].get_tag(test.TEST_FRAMEWORK) == FRAMEWORK
+                assert spans[i].get_tag(test.FRAMEWORK_VERSION) == _get_runtime_and_os_metadata()[RUNTIME_VERSION]
+                assert spans[i].get_tag(SPAN_KIND) == KIND
+                assert spans[i].get_tag(COMPONENT) == COMPONENT_VALUE
+                assert spans[i].get_tag(SESSION_ID) == str(test_session_span.span_id)
+                assert spans[i].get_tag(test.COMMAND) == "python -m unittest"
+                assert spans[i].get_tag(test.TEST_STATUS) == test.Status.SKIP.value
+                assert spans[i].get_tag(ERROR_MSG) is None
+                assert spans[i].get_tag(ERROR_TYPE) is None
+
+            assert spans[3].name == TEST_OPERATION_NAME
+            assert spans[3].get_tag(test.NAME) == "test_will_be_skipped_first"
+            assert spans[3].get_tag(test.SKIP_REASON) == "demonstrating skipping with a reason"
+            assert spans[3].get_tag(test.MODULE) == "tests.contrib.unittest_plugin.test_unittest"
+            assert spans[3].get_tag(test.SUITE) == "UnittestExampleTestCase"
+            assert spans[3].get_tag(MODULE_ID) == str(test_module_span.span_id)
+            assert spans[3].get_tag(SUITE_ID) == str(test_suite_span.span_id)
+
+            assert spans[4].name == TEST_OPERATION_NAME
```

Comment:
These big blocks of assertions are duplicated a few times. Consider deduplicating them with a function.

Reply:
I agree that they're long and duplicated, but IMO, debugging failing tests is so much simpler when you have a specific line to go to (I don't even like the `for` loop all that much).

I've looked at the [test span class](https://github.com/DataDog/dd-trace-py/blob/1.x/tests/utils.py#L596), and thought about replacing these asserts with test spans and using `span.matches()`, but I think it would not end up being much shorter or less unwieldy.

Reply:
I agree with @romainkomorndatadog. I was debugging my tests earlier and found that the `for` loop-type tests are much harder since you do not know which span failed.